controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -163,6 +164,7 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 	if err != nil {
 		return controller.RequeueWithError(fmt.Errorf("failed to update snapshot test status: %w", err))
 	}
+	metrics.RegisterSEBFailedDeployment()
```

Comment:
Should we move this to the end of this function, somewhere around [here](https://github.com/redhat-appstudio/integration-service/blob/9808d7bd67318fcad2069e2c2d130fa003119dbc/controllers/binding/snapshotenvironmentbinding_adapter.go#L194)  to ensure that we don't register the failed deployment more than once in case there is an error when cleaning up the environment and we get a requeue?

Reply:
Agreed, moved to the suggested place.